packages/types/src/cart/workflows.ts

```diff
@@ -65,3 +65,14 @@ export interface AddToCartWorkflowInputDTO {
   items: CreateCartCreateLineItemDTO[]
   cart: CartDTO
 }
+
+export interface UpdateCartWorkflowInputDTO {
+  id: string
+  promo_codes?: string[]
+  region_id?: string
```

Comment:
Can you even update the region of a cart? I am still exploring what is a possibility or not so bear with me :D

The same question on customer_id, sales_channel_id and currency_code I guess.

Reply:
Good question 😄 I'm not 100% sure either.

From the customer_id perspective, I think its possible that you'd want to move an anonymous customer to a customer thats logged in mid cart. I suppose a customer could also update region/currency, but that should have some consequences on the items that are already present in the cart, which is not done in this PR.

Reply:
@riqwan I think adding some rail guards is still part of the module's responsibility, so we should either not allow eg. modifying a region, or ensure that if one does do that, we bring the cart into a consistent state (updating what items are in the cart, etc.). I guess we could shift the responsibility to a workflow, but this seems like a core responsibility of the cart.

Anyway, these are discussions that can happen later on as well, so it's not a blocker for sure, it's all about where responsibilities lie ultimately.

Reply:
Yup, completely agree there.